MetaMask/metamask-mobile
# feat: Snaps execution refactor

## **Description**

This PR adds a refactored Snaps Webview approach and embedded execution environment to allow mobile app consume the new mobile snaps architecture.

## **Related issues**

Fixes: N/A 

## **Manual testing steps**

First --> **Copy snaps_controllers files to metamask-mobile repo** 

1. Clone [snaps monorepo](https://github.com/MetaMask/snaps) and run `yarn` on root dir.
2. navigate to `packages/snaps-controller` and run: `yarn build:source && yarn build`
3. open the `dist` directory and copy the `service` folder from `cjs`, `esm`, and `types` and paste inside `node_modules/@metamask/snaps-controller` (cjs, esm, and types) on `metamask-mobile` repo.

 Second --> Start the execution environment locally
1. Inside the snaps monorepo cloned above, navigate to: packages/snaps-execution-environments.
2. run: `yarn build:lavamoat:policy && yarn build:lavamoat && yarn build:source && yarn build`
3. Modify `scripts/start.js` from `const PUBLIC = path.join(ROOT, '/dist/browserify/iframe');` to `const PUBLIC = path.join(ROOT, '/dist/browserify/webview');`
4. run: yarn start

Third:

1. Modify `app/lib/snaps/SnapsExecutionWebView.tsx` webview source to point to `http://localhost:6363`.
2. Open the mobile app
6. Navigate to explore session and access snaps example page
7. add any snap
8. go to snaps item on settings
9. check if the installed snap is listed.

https://github.com/MetaMask/metamask-mobile/assets/44679989/808f03d6-e6cd-4652-a1c3-615b77d2f479

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**


## **CI Related Section: DO NOT EDIT**

Flag used by Bitrise to either run or skip E2E smoke tests

- Bitrise Flag:[skip ci]
